File: app/src/main/java/org/evcs/android/model/PaginatedResponse.java
package org.evcs.android.model;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class PaginatedResponse<T> {

    private int currentPage;
    private int perPage;
    private int totalPages;
    private int totalCount;
    @SerializedName("page") private List<T> mList;

    public List<T> getPage() {
        return mList;
    }

    public int getTotalPages() {
        return totalPages;
    }
}

File: app/src/main/java/org/evcs/android/model/PaginatedResponse.kt
package org.evcs.android.model

import com.google.gson.annotations.SerializedName

class PaginatedResponse<T> {
    private val currentPage = 0
    private val perPage = 0
    val totalPages = 0
    private val totalCount = 0

    @SerializedName("page")
    val page: List<T>? = null
}